src/Core/Domain/CatalogPriceRule/QueryResult/CatalogPriceRuleList.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\QueryResult;
+
+/**
+ * Transfer SpecificPrice list data
+ */
+class CatalogPriceRuleList
+{
+    /**
+     * @var CatalogPriceRuleForListing[]
+     */
+    private $catalogPriceRules;
+
+    /**
+     * @var int
+     */
+    private $totalCatalogPriceRulesCount;
```

Comment:
totalCatalogPriceRulesCount => catalogPriceRulesCount is shorter and at least explicit ;-)

Reply:
Actually I wouldn't mind making it shorter with `totalCount` maybe, but the `total` part is far from being optional, if you only use won't then what does it refer to? The count of the page or the count of all the list?

The page count is not useful for pagination, we need to total one to be able to deduce the number of pages, so I think adding this `total` prefix actually leaves out the ambiguity 🤔